File: apple/Sources/ReactNativeWatchOS/Modifiers/Grayscale.swift
import SwiftUI

/// SwiftUI `.grayscale(_:)`. Desaturates by `value`; `0` is unchanged,
/// `1` is fully gray.
enum RNWGrayscaleModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("grayscale") { view, params, _ in
            AnyView(view.grayscale(params.double("value") ?? 0))
        }
    }
}

File: src/modifiers/grayscale.ts
import { createModifier } from './createModifier';

/// SwiftUI `.grayscale(_:)`. Desaturates the view by `value`, where `0`
/// leaves colors unchanged and `1` is fully gray.
export function grayscale(value: number) {
  return createModifier('grayscale', { value });
}
